### ISSUE TITLE
Upload build assets and generate partial build manifest in the installer legs

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/runtime#44842]
We could save 7-10 minutes in our build by uploading build assets and generating a partial build manifest directly in the installer legs instead of in the "Prepare Signed Artifacts" leg which only runs after the whole build stage is done. By doing that we could shrink the "Prepare Signed Artifacts" leg down to only merge the partial build manifests and push the BuildId to BAR.

@jkoritzinsky is looking into if we need tooling support for this.

cc @mmitche @NikolaMilosavljevic @safern

**Comment by hoyosjs** [comment]
Is the time saving from this mostly download + upload time for the artifacts?

**Comment by jkoritzinsky** [comment]
Yes. And if we can remove the job, it would save us time cloning the repo as well.

**Comment by hoyosjs** [comment]
@jkoritzinsky uploading to the bar already merges the manifests into one. So maybe this could even . The only thing to verify is PB and in-build signing work/end up in the manifest.

**Comment by jkoritzinsky** [comment]
I was looking through the aspnetcore build and noticed that. That should make this much easier. I'm just trying to figure out how their repo does it since they explicitly exclude generating binlogs in the official build.

**Comment by jkoritzinsky** [comment]
I think just passing the `-publish` flag will work, but I need to figure out how to ensure that all of the uploaded manifests have unique names.

**Comment by jkoritzinsky** [comment]
They just explicitly pass the `AssetManifestFileName` property with a preset name. We should be able to do the same thing with our jobs.

**Comment by hoyosjs** [comment]
> I was looking through the aspnetcore build and noticed that. That should make this much easier. I'm just trying to figure out how their repo does it since they explicitly exclude generating binlogs in the official build.

Yeah, that always makes their's more... fun...


> I think just passing the `-publish` flag will work, but I need to figure out how to ensure that all of the uploaded manifests have unique names.

We can use the job name as the manifest name. It's guaranteed to be unique.

**Comment by ViktorHofer** [comment]
Sounds like we could get rid of the "Prepare Signed Artifacts" leg entirely. That would be fantastic.

**Comment by safern** [comment]
As a note... we could also save some time if Installer didn't depend on All Configurations leg, and in the All Configurations leg we could produce the manifest for libraries OOB packages.

**Comment by jkoritzinsky** [comment]
Installer doesn't depend on AllConfigurations any more. I agree with Santi that we should have AllConfigurations produce the manifest for the libraries OOB packages.

**Comment by safern** [comment]
> Installer doesn't depend on AllConfigurations any more.

That's great to know!

**Comment by ViktorHofer** [comment]
@tkapin @lukas-lansky @riarenas as briefly mentioned in the last "Harmonize Arcade and Runtime" meeting, we currently have [custom publishing logic](https://github.com/dotnet/runtime/tree/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/installer/publish) in place which we would like to get rid of, which should save 5-15 minutes in our official builds.
I hope we can prioritize this task as its cost shouldn't be too high but its positive impact definitely is.

**Comment by ViktorHofer** [comment]
@riarenas have you had time to look into this further part of the "Runtime <-> Arcade harmonization" effort?

**Comment by riarenas** [comment]
@ViktorHofer Yep, this is definitely still top of mind for me, I'm currently in our FR rotation until 5/14, and I will go back to it once I'm back to working on the harmonization effort.

**Comment by ViktorHofer** [comment]
Awesome, thanks for the update Ricardo.

**Comment by ViktorHofer** [comment]
Keeping this in 6.0 as @riarenas is currently working on it.

**Comment by NikolaMilosavljevic** [comment]
[Triage] @ViktorHofer is this still an issue?

**Comment by hoyosjs** [comment]
In-build signing and other legs with complex behavior (crossdac) took a dependency on this. It's not blocking for 7 - it would save us time but all legs would need to adjust appropriately.

**Comment by NikolaMilosavljevic** [comment]
> In-build signing and other legs with complex behavior (crossdac) took a dependency on this. It's not blocking for 7 - it would save us time but all legs would need to adjust appropriately.

Thanks @hoyosjs - moving to 8.0.0 milestone.

**Comment by NikolaMilosavljevic** [comment]
[Triage] @ViktorHofer is this still relevant? Should this be moved to a different repo, like arcade services?

**Comment by jkoritzinsky** [comment]
I think this is something we'd still like to investigate and this is right repo for it.
